### PR TITLE
DRY up lint-proto step in CI by reusing make target

### PR DESCRIPTION
**Pull request by krancour**
This PR will help keep developer-facing make targets and their CI counterparts in parity by reusing the `lint-proto` target in CI instead of redefining the same steps.

Without this, it could be too easy in the future to make changes to one and forget about the other.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -70,14 +70,12 @@ jobs:
...
-      - uses: bufbuild/buf-setup-action@v1
+      - name: Install linter
+        uses: bufbuild/buf-setup-action@v1
         with:
           version: "1.20.0"
-      - uses: bufbuild/buf-lint-action@v1
-        with:
-          input: api
+      - name: Run linter
+        run: make lint-proto
```
Comment (devholic) [inline .github/workflows/ci.yaml:78]: The reason I used `buf-lint-action` was to annotate message to the lines with error 🤔 (Same as the golangci-lint)

I agree with your concerns - How about adding `make lint-proto` step at the end for make sure the job fails when the one of `Makefile` known input fails?

Comment (krancour) [reply .github/workflows/ci.yaml:78]: You've reminded me... Go linting step needs this same improvement.

Not sure I entirely understood your comment. Can I propose that you are welcome to amend this PR with your suggestion?

Comment (devholic) [reply .github/workflows/ci.yaml:78]: Sure!

**Comment by devholic** [comment]
@krancour I make `--error-format` configurable to print GitHub actions formatted error in CI environment

image

**Review by devholic — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -24,7 +25,7 @@ lint-go:
 .PHONY: lint-proto
 lint-proto:
 	go mod vendor
-	buf lint api
+	buf lint api --error-format=$(BUF_LINT_ERROR_FORMAT)
```
Comment (krancour) [inline Makefile:28]: I see what you did. I honestly had no idea that the Action was using a different/specialized error format!

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -70,14 +70,12 @@ jobs:
...
-      - uses: bufbuild/buf-setup-action@v1
+      - name: Install linter
+        uses: bufbuild/buf-setup-action@v1
         with:
           version: "1.20.0"
-      - uses: bufbuild/buf-lint-action@v1
-        with:
-          input: api
+      - name: Run linter
+        run: make lint-proto BUF_LINT_ERROR_FORMAT=github-actions
```
Comment (krancour) [inline .github/workflows/ci.yaml:78]: Super-nit: I find this slightly more readable, but I know this is personal preference. If you love it the way it is, please leave it. 100% your call.

```suggestion
      - name: Run linter
        env:
          BUF_LINT_ERROR_FORMAT: github-actions
        run: make lint-proto
```

Comment (devholic) [reply .github/workflows/ci.yaml]: Thanks for the suggestion! :)

**Pull request merged**